src/monaco.contribution.ts

```diff
@@ -46,21 +57,46 @@ export class LanguageServiceDefaultsImpl implements monaco.languages.typescript.
 		}
 
 		if (this._extraLibs[filePath]) {
-			throw new Error(`${filePath} already a extra lib`);
+			this._extraLibs[filePath].version++;
+			this._extraLibs[filePath].content = content;
+		} else {
+			this._extraLibs[filePath] = {
+				content: content,
+				version: 1
+			};
+		}
+		if (this._eagerExtraLibSync) {
+			this.syncExtraLibs();
 		}
-
-		this._extraLibs[filePath] = content;
-		this._onDidChange.fire(this);
 
 		return {
 			dispose: () => {
-				if (delete this._extraLibs[filePath]) {
-					this._onDidChange.fire(this);
+				if (delete this._extraLibs[filePath] && this._eagerExtraLibSync) {
+					this.syncExtraLibs();
 				}
 			}
 		};
 	}
 
+	async syncExtraLibs() {
+		try {
+			let worker;
+			// we don't care if the get language worker fails.

```

Comment:
Not sure I follow why this is an acceptable state.

Reply:
we don't care if the get language worker fails.
This happens if addExtraLib is called before the worker has initialized. However, when the worker has finished downloading and initializes, it does so with the latest extraLibs so no sync issue can appear